synapse/storage/events.py

```diff
@@ -610,6 +594,42 @@ def event_dict(event):
             ],
         )
 
+        if event.type == EventTypes.Redaction and event.redacts is not None:
```

Comment:
is this acting on the last `event` in the loop at line 577 deliberately? needs explanation if so